src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -18,27 +18,90 @@
 
 import com.ebivariation.contigalias.dao.AssemblyDao;
 import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.entities.ChromosomeEntity;
+import com.ebivariation.contigalias.repo.AssemblyRepository;
+import org.hibernate.LazyInitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Optional;
 
 @Service
 public class AssemblyService {
 
-    private final AssemblyDao assemblyDao;
+    private final AssemblyRepository repository;
+
+    private final AssemblyDao dao;
+
+    private final Logger logger = LoggerFactory.getLogger(AssemblyService.class);
 
     @Autowired
-    public AssemblyService(@Qualifier("ftpDao") AssemblyDao assemblyDao) {
-        this.assemblyDao = assemblyDao;
+    public AssemblyService(AssemblyRepository repository, AssemblyDao dao) {
+        this.repository = repository;
+        this.dao = dao;
+    }
+
+    public Optional<AssemblyEntity> getAssemblyOrFetchByAccession(String accession) {
+        Optional<AssemblyEntity> assembly = getAssemblyByAccession(accession);
+        if (assembly.isPresent()) {
+            return assembly;
+        }
+        try {
+            Optional<AssemblyEntity> fetchAssembly = dao.getAssemblyByAccession(accession);
+            if (fetchAssembly.isPresent()) {
+                AssemblyEntity asm = fetchAssembly.get();
+                insertAssembly(asm);
+                List<ChromosomeEntity> chromosomes = asm.getChromosomes();
+                if (chromosomes != null) {
+                    chromosomes.forEach(chr -> chr.setAssembly(null));
+                }
+                return fetchAssembly;
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
why not letting the exception raise? if it reaches the controller it will send a 500 code, letting the user know that somethings wrong. Otherwise you can not know if the assembly does not exist or if something crashed.